Feature/fluxc photo chooser fixes

nbradbury opened the pull request:
This PR addresses a few issues with the already-merged photo chooser code:

* Removes the hard-coded ties to `EditPostActivity` from the photo chooser fragment, making it more reusable down the road.

* Adds a faded overlay atop to the editor to effectively disable it when the chooser is showing. More work will need to be done to better integrate the editor with the chooser, but this at least prevents the user from tapping the editor and showing the keyboard when the chooser is active.

* Refreshes the photo chooser each time it's shown so changes (photos added/deleted) are correctly reflected. Note that in 9a530ea I attempted to use a `ContentObserver` to detect changes, but that proved unreliable.

* Drops `INTERNAL_CONTENT_URI` and Scans only `EXTERNAL_CONTENT_URI` (since `INTERNAL_CONTENT_URI` is intended only for system apps).

* Delays populating the photo chooser until the user chooses to show it (previously it was populated when `EditPostActivity` was displayed, but that seemed an unnecessary memory burden).

cc: @mzorz

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:
```diff
@@ -561,6 +564,51 @@ public void hidePhotoChooser() {
+     * called by PhotoChooserFragment when user selects a single media uri
+     */
+    @Override
+    public void onPhotoChooserMediaChosen(@NonNull Uri uri) {
```
Is there a reason to have this? a particular case of  `onPhotoChooserMediaListChosen(@NonNull List<Uri> uriList)` - if not it seems it could all be handled by  this latter one and get a list of one item, maybe renaming it to `onPhotoChooserMediaChosen(@NonNull List<Uri>)`. Unless you feel strong about having a separate method for one item selections.

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:
```diff
@@ -470,7 +464,7 @@ protected void onSaveInstanceState(Bundle outState) {
 
     @Override
-    public void onConfigurationChanged(Configuration newConfig) {
+    public void  onConfigurationChanged(Configuration newConfig) {
```
extra space here

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/posts/photochooser/PhotoChooserFragment.java:
```diff
@@ -66,49 +73,35 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             @Override
             public void onClick(View v) {
-                handleIconClicked(PhotoChooserIcon.ANDROID_CAMERA);
+                mListener.onPhotoChooserIconClicked(PhotoChooserIcon.ANDROID_CAMERA);
```
I know you left this here as a trap ;) (jk)
Is there a possibility `mListener` is null here? I can see this is populated [here](https://github.com/wordpress-mobile/WordPress-Android/pull/5277/files#diff-0c023198108318246102abd97075dba2R101) in `onAttach` and it seems natural there's this pre-condition (that the fragment _has_  to be attached) in order for a click event to ever happen, but I'd add the null check just in case.

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/posts/photochooser/PhotoChooserFragment.java:
```diff
@@ -66,49 +73,35 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             @Override
             public void onClick(View v) {
-                handleIconClicked(PhotoChooserIcon.ANDROID_PICKER);
+                mListener.onPhotoChooserIconClicked(PhotoChooserIcon.ANDROID_PICKER);
```
same here

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/posts/photochooser/PhotoChooserFragment.java:
```diff
@@ -66,49 +73,35 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             @Override
             public void onClick(View v) {
-                handleIconClicked(PhotoChooserIcon.WP_MEDIA);
+                mListener.onPhotoChooserIconClicked(PhotoChooserIcon.WP_MEDIA);
```
same here

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/posts/photochooser/PhotoChooserFragment.java:
```diff
@@ -133,12 +126,10 @@ private boolean isBottomBarShowing() {
-            EditPostActivity activity = getEditPostActivity();
-            activity.addMedia(mediaUri);
-            activity.hidePhotoChooser();
+            mListener.onPhotoChooserMediaChosen(mediaUri);
```
same here

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/posts/photochooser/PhotoChooserFragment.java:
```diff
@@ -270,7 +263,7 @@ public boolean onPrepareActionMode(ActionMode mode, Menu menu) {
             if (item.getItemId() == R.id.mnu_confirm_selection) {
                 ArrayList<Uri> uriList = getAdapter().getSelectedURIs();
-                addMediaList(uriList);
+                mListener.onPhotoChooserMediaListChosen(uriList);
```
and here

nbradbury commented:
> extra space here

Fixed in 2051395

> it seems it could all be handled by this latter one and get a list of one item

Makes sense - changed in 05cc747

> Is there a possibility mListener is null here? 

It shouldn't be possible for the listener to be null, but I agree it's worth checking for. Addressed in 1c38691

mzorz commented:
:shipit: 🔥
